### ISSUE TITLE
Expose all protocols in AFM rules

### ISSUE THREAD

**Issue by jussisjostrom** [F5Networks/f5-appsvcs-extension#364]
### Is your feature request related to a problem? Please describe.
In AFM Firewall policies, rules for matching portless protocols ICMP, IPSEC/ESP, IPSEC/AH and GRE can be done, example tmsh config for one rule:
```
Allow_GRE {
            action accept
            ip-protocol gre
            destination {
                addresses {
                    2000::/3 { }
                }
            }
            source {
                address-lists {
                    /Common/SOURCENETS
                }
                vlans {
                    /Common/INGRESSVLAN
                }
            }
        }
```
But the AS3 schema for Firewall_Rule allows only "tcp", "udp" or "any" for the protocol field, so it is not possible to create granular rules matching the above-mentioned port-less protocols.

### Describe the solution you'd like
I would like to see feature parity between tmsh and AS3, regarding Firewall_Rule objects, with regards to matching protocols.

### Describe alternatives you've considered
Alternative would be to just enable protocol any, but that is not desirable for most cases where such granular rules currently exist created using tmsh and configuration is to be migrated to AS3. 
Another alternative could be to create the Firewall rules by calling directly the iControlREST. But since all the other configurations are created using AS3, it is undesirable to split up configurations to multiple methods.

### Additional context
For ICMP, it would also be desired to be able to specify the specific ICMP-codes, example tmsh config:
```
       Allow_ICMP {
            action accept
            ip-protocol ipv6-icmp
            destination {
                addresses {
                    2000::/3 { }
                }
            }
            icmp {
                1 { }
                2 { }
                3 { }
                4 { }
                128 { }
                129 { }
            }
            source {
                address-lists {
                    /Common/SOURCENETS
                }
                vlans {
                    /Common/INGRESSVLAN
                }
            }
        }
```

**Comment by dstokesf5** [comment]
Thank you for your feedback. Due to the age of this feature request, I would like to know if this is still something you need or are interested in.

**Comment by jussisjostrom** [comment]
This issue is still relevant despite it is some time since I requested it and I hope it could be implemented. When doing automation using AS3, it is quite normal to expect to be able to do same config as is possible in tmsh.

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added AUTOTOOL-2636 to our internal product backlog to expand the allowed protocols list.

**Comment by dstokesf5** [comment]
This has been resolved in the 3.32.0 release.
